Assets/MixedRealityToolkit.Tests/PlayModeTests/ConstraintTests.cs

```diff
@@ -0,0 +1,548 @@
+            manipHandler.ManipulationType = ObjectManipulator.HandMovementType.OneHanded;
+
+            var constraint = manipHandler.EnsureComponent<MoveAxisConstraint>();
+            constraint.UseLocalSpaceForConstraint = false;
+
+            yield return new WaitForFixedUpdate();
+            yield return null;
+
+            float moveAmount = 1.5f;
+            const int numHandSteps = 1;
+
+            // Hand pointing at middle of cube
+            Vector3 initialHandPosition = new Vector3(0.044f, -0.1f, 0.45f);
+            TestHand hand = new TestHand(Handedness.Right);
+            yield return hand.Show(initialHandPosition);
+
+            yield return hand.SetGesture(ArticulatedHandPose.GestureId.Pinch);
+
+            // Constrain x axis
+            constraint.ConstraintOnMovement = AxisFlags.XAxis;
+
+            yield return hand.Move(Vector3.one * moveAmount, numHandSteps);
+            yield return null;
+
+            Assert.AreEqual(initialObjectPosition.x, testObject.transform.position.x);
+            Assert.AreNotEqual(initialObjectPosition.y, testObject.transform.position.y);
+            Assert.AreNotEqual(initialObjectPosition.z, testObject.transform.position.z);
+
+            yield return hand.MoveTo(initialHandPosition, numHandSteps);
+            yield return null;
+
+            // Constrain y axis
+            constraint.ConstraintOnMovement = AxisFlags.YAxis;
+
+            yield return hand.Move(Vector3.one * moveAmount, numHandSteps);
+            yield return null;
+
+            Assert.AreNotEqual(initialObjectPosition.x, testObject.transform.position.x);
+            Assert.AreEqual(initialObjectPosition.y, testObject.transform.position.y);
+            Assert.AreNotEqual(initialObjectPosition.z, testObject.transform.position.z);
+
+            yield return hand.MoveTo(initialHandPosition, numHandSteps);
+            yield return null;
+
+            // Constrain z axis
+            constraint.ConstraintOnMovement = AxisFlags.ZAxis;
+
+            yield return hand.Move(Vector3.one * moveAmount, numHandSteps);
+            yield return null;
+
+            Assert.AreNotEqual(initialObjectPosition.x, testObject.transform.position.x);
+            Assert.AreNotEqual(initialObjectPosition.y, testObject.transform.position.y);
+            Assert.AreEqual(initialObjectPosition.z, testObject.transform.position.z);
+
+            yield return hand.MoveTo(initialHandPosition, numHandSteps);
+            yield return null;
+
+            // Constrain two axes
+            constraint.ConstraintOnMovement = AxisFlags.XAxis | AxisFlags.ZAxis;
+
+            yield return hand.Move(Vector3.one * moveAmount, numHandSteps);
+            yield return null;
+
+            Assert.AreEqual(initialObjectPosition.x, testObject.transform.position.x);
+            Assert.AreNotEqual(initialObjectPosition.y, testObject.transform.position.y);
+            Assert.AreEqual(initialObjectPosition.z, testObject.transform.position.z);
```

Comment:
does it work to set all three axis as constraints ?

Reply:
Yeah, you won't be able to move the object. I figured that this generally isn't behaviour we expect people to use, so testing it seemed unnecessary.

Reply:
I'd add the couple of lines to test this bc the stuff that breaks is usually the unexpected use cases :) . doesn't harm to have it there if it's working and we support it